lib/brexit_checker/criteria/filter.rb

```diff
@@ -0,0 +1,32 @@
+class BrexitChecker::Criteria::Filter
+  def call(selected_criteria)
+    loop do
+      old_size = selected_criteria.size
+      selected_criteria &&= possible_criteria(selected_criteria)
+      break if selected_criteria.size == old_size
```

Comment:
Ideally yes, since that would be clearer. I think we'd have to duplicate the list to do that, though, which seemed overkill. The crucial thing is that line 5 cannot add any new elements to `selected_criteria`. Do you think it needs to be clearer?

Reply:
Hmm, whilst I think we can be confident right now we're not adding new elements, it would be safer to do the correct check without the assumption to prevent any bugs being introduced in the future.

Tried a few versions to see readability:
```suggestion
      old_criteria = selected_criteria.dup
      selected_criteria &= possible_criteria(selected_criteria)
      break if selected_criteria.difference(old_criteria).any?
```

```suggestion
      old_criteria = selected_criteria.to_set
      selected_criteria &= possible_criteria(selected_criteria)
      break if selected_criteria.to_set == old_criteria.to_set
```

Reply:
Discussed in person - we add assertion to catch growing sets and potential endless loops. Will write tests to cover the assumption that new selected criteria will be a subset of the old criteria.

Reply:
I've added a 'raise' assertion (forgot that `assert` is a test-specific/Java thing) in a separate commit, which documents why it's there in case it surprises someone in future.